src/conditions/check_fields.rs

```diff
@@ -94,19 +99,23 @@ impl ContainsPredicate {
         match arg {
             CheckFieldsPredicateArg::String(s) => Ok(Box::new(Self {
                 target: target.into(),
-                arg: s.clone(),
+                arg: vec![s.clone()],
             })),
-            _ => Err("contains predicate requires a string argument".to_owned()),
+            CheckFieldsPredicateArg::VecString(ss) => Ok(Box::new(Self {
+                target: target.into(),
+                arg: ss.clone(),
+            })),
+            _ => Err("contains predicate requires a string or list of string argument".to_owned()),
         }
     }
 }
 
 impl CheckFieldsPredicate for ContainsPredicate {
     fn check(&self, event: &Event) -> bool {
         match event {
-            Event::Log(l) => l
-                .get(&self.target)
-                .map_or(false, |v| v.to_string_lossy().contains(&self.arg)),
+            Event::Log(l) => l.get(&self.target).map_or(false, |v| {
+                self.arg.iter().any(|s| v.to_string_lossy().contains(s))
+            }),
```

Comment:
I may have missed this before. This, and all the following loops, ends up running the UTF-8 check in `v.to_string_lossy`, which can be an expensive operation, once for each target string `s`. Please move that outside of the `iter`.

```suggestion
            Event::Log(l) => l.get(&self.target).map_or(false, |v| {
                let v = v.to_string_lossy();
                self.arg.iter().any(|s| v.contains(s))
            }),
```

Reply:
Ah good one - yeah I'll change that.